js/apps/admin-ui/src/groups/components/MoveDialog.tsx

```diff
@@ -12,23 +12,12 @@ type MoveDialogProps = {
 };
 
 const moveToRoot = async (source: GroupRepresentation) => {
-  await adminClient.groups.del({ id: source.id! });
-  const { id } = await adminClient.groups.create({
-    ...source,
-    id: undefined,
-  });
-  if (source.subGroups) {
-    await Promise.all(
-      source.subGroups.map((s) =>
-        adminClient.groups.setOrCreateChild(
-          { id: id! },
-          {
-            ...s,
-            id: undefined,
-          }
-        )
-      )
-    );
+  try {
+    await adminClient.groups.create(source);
+  } catch (error: any) {
+    if (error.response) {
```

Comment:
The `adminClient.groups.create()` function call throws a `new Error('location header is not found in request: ${res.url}')` exception. Without the try-catch, this exception would be unhandled here resulting in an error shown to the user. The same exception is thrown in the functions called by the "moveToGroup" function, so I modeled the exception handling after it. I believe the intent is for that specific exception to be caught and ignored, while exceptions resulting from HTTP requests being caught and rethrown. In both cases, it does not matter that a location header is not present in the response, as that is only used to retrieve the created group Id, which doesn't really happen because the existing group resource is simply modified.

Reply:
@edewit I believe you wrote this code originally, there must be a better way to handle this, no? I feel like this exception should not be thrown in the first place if the request does complete with the intended effect.

Reply:
it's handled also in the same file